astropy/utils/shapes.py

```diff
@@ -241,9 +241,15 @@ def __array_function__(self, function, types, args, kwargs):
             # since classes could override the method.
             name = {np.amax: 'max',
                     np.amin: 'min'}.get(function, function.__name__)
+
             method = getattr(self, name, None)
             if method is not None:
-                return method(*args[1:], **kwargs)
+                if callable(method):
+                    return method(*args[1:], **kwargs)
+                elif (len(args) == 1 and hasattr(
+                        getattr(self.__class__, name, None), '__get__')):
+                    # Descriptor/property; assume it holds the result.
+                    return method
```

Comment:
You probably need a final `else` block to catch something unexpected. Not sure how to get there, but at least having some kind of exception here that points right here to an unforeseen situation.

Reply:
Right now it just falls through to executing the numpy function as is, but it would make sense to insert a comment, to make clear that that is on purpose.

Looking at this again, I wonder if I shouldn't just be more explicit and just check for `np.shape`, `np.ndim` and `np.size` (but still check that a method is callable - that was definitely an oversight).

Reply:
My initial thought was to add an `_APPLICABLE_PROPERTIES` list to go with `_APPLICABLE_FUNCTIONS`...